src/main/scala/cromwell/engine/db/slick/WorkflowExecutionComponent.scala

```diff
@@ -37,14 +37,30 @@ trait WorkflowExecutionComponent {
       workflowExecutionUuid, unique = true)
   }
 
-  val workflowExecutions = TableQuery[WorkflowExecutions]
+  protected val workflowExecutions = TableQuery[WorkflowExecutions]
 
   val workflowExecutionsAutoInc = workflowExecutions returning workflowExecutions.
     map(_.workflowExecutionId) into ((a, id) => a.copy(workflowExecutionId = Some(id)))
 
-  val workflowExecutionByID = Compiled(
-    (id: Rep[Int]) => for {
+  val workflowExecutionsByWorkflowExecutionUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
       workflowExecution <- workflowExecutions
-      if workflowExecution.workflowExecutionId === id
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
     } yield workflowExecution)
+
+  val workflowExecutionStatusesByWorkflowExecutionUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
+      workflowExecution <- workflowExecutions
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
+    } yield workflowExecution.status)
+
+  // NOTE: No precompile for you!
+  // [Compile] works for all functions ... consisting only of individual columns
```

Comment:
tl;dr I think the current code is fine.  :smile:  Perhaps the comment might also say why this is particularly not a problem given our use case?

Excellent link, what Vogt says there is IMHO a lot more clear than the actual docs.  His follow-on also makes sense:

>You can implement this yourself by the way, by writing a function, 
which takes the set and compiles and caches different queries depending 
on the arity. 

That got me thinking about the use case for this method, and it turns out this is currently executed only once at server restart:

```
workflowInfos <- dataAccess.getWorkflowsByState(Seq(WorkflowSubmitted, WorkflowRunning))
```

So if we narrowed the API to only get Submitted/Running workflows this should be compilable, but given that the query executes only once at startup there doesn't currently seem to be a point to compilation.